Simplenote/src/androidTest/java/com/automattic/simplenote/LoginWithCorrectPasswordLogoutTest.java

```diff
@@ -0,0 +1,182 @@
+
+        ViewInteraction pwField = onView(
+                allOf(childAtPosition(
+                        childAtPosition(
+                                withId(R.id.input_password),
+                                0),
+                        0),
+                        isDisplayed()));
+        pwField.perform(replaceText("*-Re7]J4Ux8q)g?X"), closeSoftKeyboard());
+
+        ViewInteraction loginButton2 = onView(
+                allOf(withId(R.id.button), withText("Log In"),
+                        childAtPosition(
+                                childAtPosition(
+                                        withId(R.id.main),
+                                        1),
+                                2),
+                        isDisplayed()));
+        loginButton2.perform(click());
+
+        try {
+            sleep(2000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        ViewInteraction openDrawer = onView(
+                allOf(withContentDescription("Open drawer"),
+                        childAtPosition(
+                                allOf(withId(R.id.toolbar),
+                                        childAtPosition(
+                                                withClassName(is("com.google.android.material.appbar.AppBarLayout")),
+                                                0)),
+                                1),
+                        isDisplayed()));
+        openDrawer.perform(click());
+
+        ViewInteraction navigationMenuItemView = onView(
+                allOf(childAtPosition(
+                        allOf(withId(R.id.design_navigation_view),
+                                childAtPosition(
+                                        withId(R.id.navigation_view),
+                                        0)),
+                        3),
+                        isDisplayed()));
+        navigationMenuItemView.perform(click());
+
+        swipeToBottom();
+
+        try {
+            sleep(2000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        onView(withText(endsWith("Log out"))).perform(click());
+
+    }
+
+
+    private static void swipe(int start, int end, int delay) {
+        long downTime = SystemClock.uptimeMillis();
+        long eventTime = SystemClock.uptimeMillis();
+        Instrumentation inst = getInstrumentation();
+
+        MotionEvent event = MotionEvent.obtain(downTime, eventTime, MotionEvent.ACTION_DOWN, 500, start, 0);
+        inst.sendPointerSync(event);
+        eventTime = SystemClock.uptimeMillis() + delay;
+        event = MotionEvent.obtain(downTime, eventTime, MotionEvent.ACTION_MOVE, 500, end, 0);
+        inst.sendPointerSync(event);
+        event = MotionEvent.obtain(downTime, eventTime, MotionEvent.ACTION_UP, 500, end, 0);
+        inst.sendPointerSync(event);
+        SystemClock.sleep(2000);
+    }
```

Comment:
Could you tell me more about this method and the benefit of using `swipeToBottom()` in this test, versus something like scrolling to a specific element that you want to interact with?

Reply:
I updated this with more swipe methods such as 

`swipeToBottom` - Swipe all the way to the bottom of the screen
`scrollSlowlyDown`- Swipe down one page at a time
`swipeToTop` - Swipe to the top
`scrollSlowlyUp`- Swipe up one page at a time


Those methods are nice to have a variety of swipe actions on automated tests, but we could also scroll automatically to the specific element

Also, was thinking instead of scrolling to the specific element to update the scroll to the `scrollSlowlyUp` where it scrolls slowly, I think would be more similar to a mainly user behaviour, any thoughts?

Reply:
> Also, was thinking instead of scrolling to the specific element to update the scroll to the scrollSlowlyUp where it scrolls slowly, I think would be more similar to a mainly user behaviour, because when going to the settings view, user doesn't see the logout option right away, I think it's a more natural user behaviour slowly scrolling until find it, any thoughts?

That's an interesting thought! What are the pros/cons you see for using a more natural scroll behavior in these tests? Are there specific risks you are trying to address with that?

Reply:
The pros would be get the more similar to the user and also try to catch any crashes within the thread while scrolling, I noticed on previous projects, if the thread was still being loaded, scrolling slowly would crash (also reproducible manually) but scrolling automatically to the specific touch point wouldn't crash on the automated test.